django/db/models/sql/query.py

```diff
@@ -65,12 +68,35 @@ def get_field_names_from_opts(opts):
     )
 
 
+def get_paths_from_expression(expr):
+    if hasattr(expr, "lhs"):
+        yield from get_paths_from_expression(expr.lhs)
+    if hasattr(expr, "rhs"):
+        yield from get_paths_from_expression(expr.rhs)
+    if isinstance(expr, F):
+        yield expr.name
+    if isinstance(expr, Func):
+        for func_expression in expr.source_expressions:
+            yield from get_paths_from_expression(func_expression)
+    if isinstance(expr, Case):
+        for case in expr.cases:
+            yield from get_children_from_q(case.condition)
```

Comment:
As far as I'm aware, we don't need to handle all kind of expressions separately. We should be able to reuse `Expression.get_source_expressions()`. For example the following works for me:
```python
def get_paths_from_expression(expr):
    if isinstance(expr, F):
        yield expr.name 
    elif isinstance(expr, Q):
        yield from get_children_from_q(expr)
    elif hasattr(expr, "resolve_expression"):
        for source_expr in expr.get_source_expressions():
            yield from get_paths_from_expression(source_expr)

```